pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+
+        Each function is named with the lower case method namd prepended with
+        'fake_'.
+        """
+        if self.verbose:
+            print('mock_imethodcall method=%s, namespace=%s, '
+                  'response_params_rqd=%s\nparams=%s'
+                  % (methodname, namespace, response_params_rqd, params))
+
+        method_name = '_fake_' + methodname.lower()
+
+        method_name = getattr(self, method_name)
+        result = method_name(namespace, **params)
+
+        if self.verbose:
+            print('mock result %s' % result)
+
+        return result
+
+    def _mock_methodcall(self, methodname, localobject, Params=None, **params):
+        # pylint: disable=invalid-name
+        """
+        Mock the execution of WBEMConnection _methodcall which is the common
+        calling interface for invokemethod
+        """
+        if self.verbose:
+            print('mock_imethodcall method=%s, namespace=%s, '
+                  'response_params_rqd=%s\nparams=%s'
+                  % (methodname, localobject, Params, params))
+
+        result = self._fake_invokemethod(methodname,
+                                         localobject,
+                                         Params=Params,
+                                         **params)
+
+        if self.verbose:
+            print('mock result %s' % result)
+
+    ################################################################
+    #
+    #   Methods to insert data into fake repository
+    #
+    ################################################################
+
+    @staticmethod
+    def display_objects(obj_type, objects_dict):
+        """
+        Display a set of objects of obj_type from the dictionary defined
+        by the parameter objects_dict. obj_type is a string that defines the
+        type of object (instance, class, qualifier declaration).
+        """
+        if not objects_dict:
+            return
+        for ns, objects in six.iteritems(objects_dict):
+            print('namespace %s: %s %s' % (ns, len(objects), obj_type))
+            for obj in six.itervalues(objects):
+                print('obj %s' % obj)
+                print(obj.tomof())
+
+    def display_repository(self):
+        """
+        Display everything in repository
+        """
+        print('===============Repository====================')
+        self.display_objects('qualifierdeclarations', self.qualifiers)
+        self.display_objects('classes', self.classes)
+
+        # instances are special because the inner struct is a list
+        for ns in six.iterkeys(self.instances):
+            print('instances namespace %s' % ns)
+            insts = self.instances[ns]
+            for inst in insts:
+                print('path=%s' % inst.path)
+                print(inst.tomof())
+        print('============End Repository=================')
+
+    def compile_mof(self, mof_file, namespace=None):
+        """
+        Compile mof from files
```

Comment:
This is a public method.
* Use a "Parameters:" section to describe its parameters
* State that the CIM objects defined in the MOF files are being added to the specified namespace in the local repo.
* Make a statement about prerequisite CIM objects, both in repo and in the MOF file.
* Describe how MOF include files are located.
* Describe which pragmas are honored.

Reply:
Parameters. Done; Namespace. Done; Added all of the parameters from MOFCompiler to this call just to keep compatibility.  We do not restrict pragmas, and include files are through the search
mechanism

Also noted that we discard upon compile failure.

DONE